internal/server/websocketServer.go

```diff
@@ -215,11 +219,78 @@ func (s *WebsocketServer) WebsocketHandler(w http.ResponseWriter, r *http.Reques
 			continue
 		}
 
-		// Write response
-		writeResponse(conn, respBytes)
+		// Write response with retries for transient errors
+		if err := WriteResponseWithRetries(conn, respBytes, config.DiceConfig.WebSocket.MaxWriteResponseRetries); err != nil {
+			s.logger.Error(fmt.Sprintf("Error reading message: %v", err))
+			break // Exit the loop on write error
+		}
 	}
 }
 
+func WriteResponseWithRetries(conn *websocket.Conn, text []byte, maxRetries int) error {
+	for attempts := 0; attempts < maxRetries; attempts++ {
+		// Set a write deadline
+		if err := conn.SetWriteDeadline(time.Now().Add(config.DiceConfig.WebSocket.WriteResponseTimeout)); err != nil {
+			slog.Error(fmt.Sprintf("Error setting write deadline: %v", err))
+			return err
+		}
+
+		// Attempt to write message
+		err := conn.WriteMessage(websocket.TextMessage, text)
+		if err == nil {
+			break // Exit loop if write succeeds
+		}
+
+		// Handle network errors
+		var netErr *net.OpError
+		if !errors.As(err, &netErr) {
+			return fmt.Errorf("error writing message: %w", err)
+		}
+
+		opErr, ok := netErr.Err.(*os.SyscallError)
+		if !ok {
+			return fmt.Errorf("network operation error: %w", err)
+		}
+
+		if opErr.Syscall != "write" {
+			return fmt.Errorf("unexpected syscall error: %w", err)
+		}
+
+		switch opErr.Err {
+		case syscall.EPIPE:
+			return fmt.Errorf("broken pipe: %w", err)
+		case syscall.ECONNRESET:
+			return fmt.Errorf("connection reset by peer: %w", err)
+		case syscall.ENOBUFS:
+			return fmt.Errorf("no buffer space available: %w", err)
+		case syscall.EAGAIN:
+			// Exponential backoff with jitter
+			backoffDuration := time.Duration(attempts+1)*100*time.Millisecond + time.Duration(rand.Intn(50))*time.Millisecond 
+
+			slog.Warn(fmt.Sprintf(
+				"Temporary issue (EAGAIN) on attempt %d. Retrying in %v...", 
+				attempts+1, backoffDuration,
+			))
+
+			time.Sleep(backoffDuration)
+			continue
+		default:
+			return fmt.Errorf("write error: %w", err)
+		}
+	}
+
+	return nil
+}
+
 func writeResponse(conn *websocket.Conn, text []byte) {
```

Comment:
Shall we just reuse `WriteResponseWithRetries(conn, respBytes, config.DiceConfig.WebSocket.MaxWriteResponseRetries);` for below case also?
```
respBytes, err := json.Marshal(responseValue)
if err != nil {
```

Reply:
I didn't use the writeResponse function with retries here since it is just returning an error and not the actual responseValue. Please let me know your thoughts and if I should replace it `WriteResponseWithRetries`?

Reply:
Sounds good.